comp/otelcol/otlp/components/pipeline/provider/provider.go

```diff
@@ -17,15 +18,28 @@ import (
 	"go.opentelemetry.io/collector/confmap/provider/httpsprovider"
 	"go.opentelemetry.io/collector/confmap/provider/yamlprovider"
 	"go.opentelemetry.io/collector/otelcol"
+	"gopkg.in/yaml.v3"
 )
 
+type ExtendedConfigProvider interface {
+	otelcol.ConfigProvider
+	GetProvidedConf() []byte
+	GetEnhancedConf() []byte
```

Comment:
Should we add error ?

Reply:
Are you talking about return an error from GetProvidedConf/ GetEnhancedConf ?
These getters just return a struct field, so I don't think an error could occur here ?